### ISSUE TITLE
Helidon Nima gRPC server stops responding after a few seconds to benchmark program

### ISSUE THREAD

**Issue by efagerho** [helidon-io/helidon#7020]
Platform: MacOS Monterey M1 chip
JDK: openjdk 20.0.1 2023-04-18

I wrote the following benchmarks:

Classic GRPC server: [https://github.com/efagerho/grpc-gradle](https://github.com/efagerho/grpc-gradle/blob/master/server/src/main/java/io/github/efagerho/loom/ServerMain.java)
Nima gRPC Server: [https://github.com/efagerho/loom-grpc-gradle](https://github.com/efagerho/loom-grpc-gradle/blob/master/server/src/main/java/io/github/efagerho/loom/ServerMain.java)
Benchmark client: [https://github.com/efagerho/loom-grpc-gradle-benchmark](https://github.com/efagerho/loom-grpc-gradle-benchmark/blob/master/benchmark/src/main/java/io/github/efagerho/loom/Benchmark.java)

When running the benchmark client against the Classic gRPC server I get:

```
% NUM_CLIENTS=1 ./gradlew :benchmark:run

> Task :benchmark:run
0 RPS
1282 RPS
6084 RPS
10177 RPS
13321 RPS
15311 RPS
15437 RPS
16199 RPS
15727 RPS
16046 RPS
16200 RPS
15964 RPS
16035 RPS
16225 RPS
...
```

Against the Nima server I get:

```
% NUM_CLIENTS=1 ./gradlew :benchmark:run

> Task :benchmark:run
0 RPS
2556 RPS
3400 RPS
0 RPS
0 RPS
0 RPS
0 RPS
0 RPS
0 RPS
0 RPS
0 RPS
...
```

In other words, it stops responding to requests. JProfiler shows almost no CPU usage on the JVM running the gRPC server when it stops responding to requests.

**Comment by efagerho** [comment]
A thread dump shows that all platform threads are parked and the gRPC I/O threads are stuck on KQueue poll(), so I'm assuming that they're waiting for a response from the Helidon Nima gRPC server.

**Comment by tomas-langer** [comment]
Attempting to reproduce with latest snapshot (we have done some work on HTTP/2)
